docs/en/migration/code.md

```diff
@@ -1 +1,137 @@
+
+- `TextDetectorMixin` has been removed since its implementation overlaps with `TextDetLocalVisualizer`.
+
+### Head
+
+- `HeadMixin`, the base class that `XXXHead` had to inherit from in version 0.x, has been replaced by `BaseTextDetHead`. `get_boundary()` and `resize_boundary()` are now rewritten as `__call__()` and `rescale()`  in `BaseTextDetPostProcessor`.
+
+### ModuleLoss
+
+- Data transforms `XXXTargets` in text detection tasks are all moved to `XXXModuleLoss._get_target_single()`. Target-related configurations are no longer specified in the data pipeline but in `XXXLoss` instead.
+
+### Postprocessor
+
+- The logic in the original `XXXPostprocessor.__call__()` are transferred to the refactored `XXXPostprocessor.get_text_instances()`.
+
+- `BasePostprocessor` is refactored to `BaseTextDetPostProcessor`. This base class splits and processes the model output predictions one by one and supports automatic scaling of the output polygon or bounding box based on `scale_factor`.
+
+## Text Recognition
+
+### Key Changes (TL;DR)
+
+- Due to the change of the character order and some bugs in the model architecture being fixed, the recognition model weights in 0.x can no longer be directly used in 1.0. We will provide a migration script and tutorial for those who need it.
+
+- The support of SegOCR has been removed. TPS-CRNN will still be supported in a later version.
+
+- Test time augmentation will be supported in the upcoming release.
+
+- *Label converter* module has been removed and its functions have been split into *Dictionary*, *ModuleLoss* and *Postprocessor*.
+
+- The definition of `max_seq_len` has been unified and now it represents the original output length of the model.
+
+### Label Converter
+
+- The original label converters had spelling errors (written as label convertors). We fixed them by removing label converters from this project.
+
+- The part responsible for converting characters/strings to and from numeric indexes was extracted to *Dictionary*.
+
+- In older versions, different label converters would have different special character sets and character order. In version 0.x, the character order was as follows.
+
+- | Converter                       | Character order                      |
+  | ------------------------------- | ------------------------------------ |
+  | `AttnConvertor`, `ABIConvertor` | <UKN>, \<BOS/EOS>, <PAD>, characters |
+  | `CTCConvertor`                  | <BLK>, <UKN>, characters             |
+
+在 1.0 中，我们不再以任务为边界设计不同的字典和字符序，取而代之的是统一了字符序的 Dictionary，其字符序为 characters, \<BOS/EOS>, , 。`CTCConvertor` 中 被等价替换为 。
+
+In 1.0, instead of designing different dictionaries and character orders for different tasks, we have a unified *Dictionary* implementation with the character order always as characters, \<BOS/EOS>, <PAD>, <UKN>. <BLK> in `CTCConvertor` has been equivalently replaced by <PAD>.
```

Comment:
```suggestion
In 1.0, instead of designing different dictionaries and character orders for different tasks, we have a unified *Dictionary* implementation with the character order always as characters, <BOS/EOS>, <PAD>, <UKN>. <BLK> in `CTCConvertor` has been equivalently replaced by <PAD>.
```

![image](https://user-images.githubusercontent.com/45810070/187414545-d5619f04-0ba6-4475-993c-47ef6b4bfb29.png)